### ISSUE TITLE
Downloading, Installing, Configuring.

### ISSUE THREAD

**Issue by ud4x0** [UnamSanctam/UnamWebPanel#198]
Hello. I have a problem with the latest version.
1. I did update & upgrade
sudo apt update
sudo apt upgrade
2. I have installed apache 2
sudo apt install apache2
3.I installed ufw and did port 80 and 433 permission
sudo ufw allow ssh
sudo ufw allow 80
sudo ufw allow 443
sudo ufw enable
4. turned on ufw.
5.I have installed php 7.4
sudo apt install php7.4 php7.4-mysql php-common php7.4-cli php7.4-json php7.4-common php7.4-opcache libapache2-mod-php7.4
6. I have installed sqlite
sudo apt install sqlite3
7. I downloaded unam web panel
sudo wget
https://github.com/UnamSanctam/UnamWebPanel/releases/tag/uwp-v1.5.0

8. I unpacked the zip
unzip uwp-v1.5.0.zip

9. I changed the password in config
10 login
11. Waited 24 hours and nothing show.

  In miner i have http: //ip/api/endpoint.php

What step i left?

**Comment by UnamSanctam** [comment]
Firstly, that's an old version (1.5.0), secondly are you sure that you have port 80 (http) open?

**Comment by ud4x0** [comment]
I just use VPS, there are all ports open TCP/UDP also in firewall

**Comment by UnamSanctam** [comment]
And if you visit http://ip/api/endpoint.php from the computer you're running the miner on then you see the endpoint page?

**Comment by ud4x0** [comment]
yeah, there is communication - >Error: Invalid data

**Comment by ud4x0** [comment]
vps is using http and miner using http protocol

**Comment by UnamSanctam** [comment]
First update your web panel and check.

**Comment by ud4x0** [comment]
ahh, i remember now why i chose 1.5. In 1.6 version i cant even login. There is communication Error, Something went wrong. When i will make all htdocs with chmod 777 login work. So what i need to increase permission?

**Comment by ud4x0** [comment]
one session i had valid, so i receive communication: DataTables warning: table id=miner-table - Ajax error. For more information about this error, please see http://datatables.net/tn/7

**Comment by UnamSanctam** [comment]
You need to make the db folder and unamwebpanel.db inside it writable. Did you do all the correct `chown` and `chmod`?

**Comment by ud4x0** [comment]
what chmod unamwebpanel.db should to have?

**Comment by ud4x0** [comment]
i didnt use chown anymore.

**Comment by UnamSanctam** [comment]
Usually you do `sudo chown -R $USER:$USER htdocslocation` and `sudo chmod -R 755 htdocslocation` and then you might also need additional permissions for the database.

**Comment by ud4x0** [comment]
database has nothing to login i think, so i didnt do anything with it yet. i have still Error problem with login after your commands

**Comment by ud4x0** [comment]
in htdocs folder i just did `sudo chown -R $USER:$USER *` then `sudo chmod -R 755 *` * - mean ALL in this folder

**Comment by ud4x0** [comment]
i see that not only me has this problem before someone warn also

**Comment by UnamSanctam** [comment]
The reason you get an error is because you have set up something wrong, check for an error.log file or for a log file inside the `__UNAM_LIB\Logs` folder.

**Comment by ud4x0** [comment]
i didnt make any additional steps than above and your suggestions. I described all step i made

**Comment by ud4x0** [comment]
so i dont understand more what is wrong?. I try to understand.  During one week i reinstalled it so many times, same problem still.

**Comment by ud4x0** [comment]
i dont have any logs or error,log in htdocs. I searched my command and checked my own

**Comment by UnamSanctam** [comment]
You only have the dummy file inside `__UNAM_LIB\Logs`? And are you sure that you have all the files? There shouldn't be any problem with setting up apache and then the web panel, I helped someone do it just yesterday. If you use Ubuntu then you can use this guide https://www.digitalocean.com/community/tutorials/how-to-install-the-apache-web-server-on-ubuntu-20-04.

**Comment by ud4x0** [comment]
I followed this guide + installed php 7.4 then sqlite3. Same problem with login

**Comment by ud4x0** [comment]
Hi Unam, I tried do it on 2 vps, still problem with login. Error something went wrong. After resolving problem i would like to make installation script for everyone, that will configure everything itself, but first I need to resolve my problem . I don't know what is wrong

**Comment by UnamSanctam** [comment]
And you tried using `sudo wget https://github.com/UnamSanctam/UnamWebPanel/releases/download/uwp-v1.6.0/UnamWebPanel.zip` to download the files?

**Comment by ud4x0** [comment]
Yeah exactly. with sudo.  Maybe I should to download by sftp, but i don't see difference

**Comment by UnamSanctam** [comment]
You downloaded that exact file? I set up a test Ubuntu VM today for the web panel and it worked fine.

**Comment by ud4x0** [comment]
Exactly this file, then unzip UnamWebPanel command. In /var/www/html

**Comment by ud4x0** [comment]
I will try install LAMPP ( xampp on Linux) then install sqlite3 and sqlite3-php. It have preconfigured settings. Maybe it will work

**Comment by ud4x0** [comment]
Hello, Unam. XAMPP on Linux works, but i needed to put command chmod 777. I tried to download database but it's forbidden. If there is any more risk, if everything in htdocs has 777 permission?

**Comment by UnamSanctam** [comment]
>Hello, Unam. XAMPP on Linux works, but i needed to put command chmod 777. I tried to download database but it's forbidden. If there is any more risk, if everything in htdocs has 777 permission?

It's generally not recommended, but if that server is only used to host the web server and nothing else then it's technically fine if you can't get it working otherwise. Though you should be able to lower the required permissions as long as the owner of the all the files is the correct user. The standard permissions used are usually 644 or 755 on the premise that the owner of the files is the user that is running Apache.
